### PR TITLE
ARROW-15350: [Doc][Guide] Add styling and linters info section

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR adds info (links to existing documentation) about styling to the New Contributor's Guide.

### CODE REVIEW

File: docs/source/developers/guide/index.rst [hunk 1]
```diff
@@ -134,7 +134,8 @@ of adding a basic feature.
 
 #. **Once you implemented the planned fix or feature, write and run tests for it**
 
-   See detailed instructions on how to :ref:`test <testing>`.
+   See detailed instructions on how to :ref:`test <testing>`. Also check the
+   :ref:`syle of the code<styling>` before proceeding to the next step!
```
Comment (thisisnic) [inline docs/source/developers/guide/index.rst:138]: It's a little unclear what it means to check the style.  Could we change this to a more direct instruction, e.g. "run the styler" or "run the linter to make sure the code is styled correctly", or something like that?

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+.. _styling:
+
+**********
+Styling 😎
+**********
+
+Each language in the Apache Arrow project follows its own style guides.
+
+In this section we will connect to the existing documentation
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:35]: "connect" is a teensy bit unclear; perhaps instead something like "provide links to" or similar?

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+to make it easier for you to find the relevant information about
+styling.
+
+.. tabs::
+
+   .. tab:: PyArrow
+
+      To checking styling issues in python files we use the
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:43]:
```suggestion
      To checking styling issues in Python files we use the
```

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      :ref:`Archery utility<archery>` tool developed within Apache
+      Arrow to ease some of the development tasks.
+
+      The instructions on how to set up and use the utility
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:47]:
```suggestion
      The instructions on how to set up and use Archery
```

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      can be found in the Coding Style section of the
+      :ref:`python-development`.
+
+   .. tab:: R package
+
+      For the R package you can use ``{lintr}`` or ``{styler}``
+      to check if the code follows the tidyverse style.
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:54]: Perhaps link the words "tidyverse style" to the tidyverse style guide.

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+      The instructions on how to use either of these two packages
+      can be found in the Styling an linting section of the
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:57]: I'd put the section name in quotes

**Review by thisisnic — COMMENTED** [review]
Just a few minor changes

File: docs/source/developers/guide/index.rst (hunk 1 above)
Comment (AlenkaF) [reply docs/source/developers/guide/index.rst:138]: Yeah, that would be better! Will do.

File: docs/source/developers/guide/step_by_step/styling.rst (hunk 2 above)
Comment (AlenkaF) [reply docs/source/developers/guide/step_by_step/styling.rst:35]: Link is better, agree. Will do.

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+   .. tab:: R package
+
+      For the R package you can use ``{lintr}`` or ``{styler}``
+      to check if the code follows the
+      `tidyverse style <https://style.tidyverse.org/>`_.
+
+      The instructions on how to use either of these two packages
+      can be found in the
+      `Styling an linting section of the Common developer workflow tasks <https://arrow.apache.org/docs/r/articles/developers/workflow.html#styling-and-linting>`_.
```
Comment (thisisnic) [inline docs/source/developers/guide/step_by_step/styling.rst:59]:
```suggestion
      `Styling and Linting section of the Common developer workflow tasks <https://arrow.apache.org/docs/r/articles/developers/workflow.html#styling-and-linting>`_.
```

**Review by thisisnic — APPROVED** [review]
One small change, otherwise LGTM

**Comment by AlenkaF** [comment]
Thank you Nic!

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+.. specific language governing permissions and limitations
+.. under the License.
+
+
+.. SCOPE OF THIS SECTION
+.. This section should include extra description to the
+.. language-specific documentation. Possible topics to
+.. add: How to run one test, test file or all the tests
+.. together and why is it good to do that.
+.. What if the unconnected tests start failing? etc.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/styling.rst:24]: This paragraph is probably from copy pasting another file?

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+to make it easier for you to find the relevant information about
+linters and styling of the code.
+
+.. tabs::
+
+   .. tab:: PyArrow
+
+      To checking styling issues in Python files we use the
+      :ref:`Archery utility<archery>` tool developed within Apache
+      Arrow to ease some of the development tasks.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/styling.rst:45]: I would maybe first mention that we use flake8 for linting, and then in a second sentence say that Archery is developed to help with this.

File: docs/source/developers/guide/step_by_step/styling.rst [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      `tidyverse style <https://style.tidyverse.org/>`_.
+
+      The instructions on how to use either of these two packages
+      can be found in the
+      `Styling and Linting section of the Common developer workflow tasks <https://arrow.apache.org/docs/r/articles/developers/workflow.html#styling-and-linting>`_.
+
+Pre-commit
+----------
+
+It is useful to setup `pre-commit <https://pre-commit.com/>`_,
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/styling.rst:64]:
```suggestion
It is useful to set up `pre-commit <https://pre-commit.com/>`_,
```

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! A few small comments

File: docs/source/developers/guide/step_by_step/styling.rst (hunk 8 above)
Comment (AlenkaF) [reply docs/source/developers/guide/step_by_step/styling.rst:24]: Good catch. Yes, will remove 👍

File: docs/source/developers/guide/step_by_step/styling.rst (hunk 9 above)
Comment (AlenkaF) [reply docs/source/developers/guide/step_by_step/styling.rst:45]: Great, will do!

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**